JavaScript/Topics/8.JSON/README.md

```diff
@@ -1,142 +1,261 @@
+
+Example:
+
+```JSON
+{
+    "firstName": "Rahul",
+    "lastName": "Roy",
+    "gender": "male",
+    "age": 24,
+    "address": {
+        "streetAddress": "MG road",
+        "city": "Kolkata",
+        "state": "West Bengal",
+        "postalCode": "95221"
+    },
+    "phoneNumbers": [
+        { "type": "home", "number": "27627" }
+    ]
+}
 ```
 
+### JSON - Evaluates to JavaScript Objects
 
-#### JSON Values and Types
-  JSON values can be of any type. JSON types are:
-  - String
-  - Number
-  - Object
-  - Array
-  - Boolean
-  - `null`
+The JSON format is almost identical to JavaScript objects. In JSON, keys must be strings, written with double quotes.
 
-### Arrays in JSON
-  JSON arrays are enclosed in square brackets `[]` and can contain any type of data.
+Example:
 
-- Example
-```json
+#### JSON
+
+```JSON
 {
-    "employees" : [
-        {
-            "name": "Bob",
-            "age": 35
-        },
-        {
-            "name": "Jack",
-            "age": 28
-        },
-        {
-            "name": "John",
-            "age": 31
-        }
+    "firstName": "Rahul",
+    "lastName": "Roy",
+    "gender": "male",
+    "age": 24,
+    "address": {
+        "streetAddress": "MG road",
+        "city": "Kolkata",
+        "state": "West Bengal",
+        "postalCode": "95221"
+    },
+    "phoneNumbers": [
+        { "type": "home", "number": "27627" }
     ]
 }
 ```
 
-#### Access array items from JSON
-Access array items from JSON using square brackets `[]` and index number.
+In JavaScript, keys can be strings, numbers, or identifier names.
+
+Example:
+
+### Javascript
+
+```js
+const person = {
+    firstName: "Rahul",
+    lastName: "Roy",
+    gender: "male",
+    age: 24,
+    address: {
+        streetAddress: "MG road",
+        city: "Kolkata",
+        state: "West Bengal",
+        postalCode: 95221
+    },
+    phoneNumbers: [
+        { type: "home", number: 27627 }
+    ]
+};
+```
 
-```javascript
-const jsonObject = {
-    "array" : ["India", "UK", "USA"]
+*Note: JSON names require double quotes.*
+
+### JSON Values
+
+In JSON, values must be one of the following data types:
+
+- string
+- number
+- object
+- array
+- boolean
+- null
+
+## JSON vs XML
+
+Both JSON and XML can be used to receive data from a web server.
+
+### JSON
+
+```JSON
+{
+     "students":
+          [
+               { "firstName":"Rahul", "lastName":"Roy" },
+               { "firstName":"Raj", "lastName":"Biswas" },
+               { "firstName":"Anjali", "lastName":"Mathur" }
+          ]
 }
+```
 
-console.log(jsonObject.array[0]); // Output: India
-console.log(jsonObject.array[1]); // Output: UK
+### XML
+
+```xml
+<students>
+  <student>
+    <firstName>Rahul</firstName> <lastName>Roy</lastName>
+  </student>
+  <student>
+    <firstName>Raj</firstName> <lastName>Biswas</lastName>
+  </student>
+  <student>
+    <firstName>Anjali</firstName> <lastName>Mathur</lastName>
+  </student>
+</students>
```

Comment:
Write XML like that

```xml
<students>
    <student>
        <firstName>Rahul</firstName>
        <lastName>Roy</lastName>
    </student>
    <student>
        <firstName>Raj</firstName>
        <lastName>Biswas</lastName>
    </student>
    <student>
        <firstName>Anjali</firstName>
        <lastName>Mathur</lastName>
    </student>
</students>
```

Reply:
@rwiteshbera Look at the suggested changes. :)

Reply:
@rwiteshbera work on these changes!!